### DISCUSSION TITLE
How to create a specific licensing scheme with updatable expiry and other criteria.

### DISCUSSION THREAD

**Discussion by ManlyBoy** [keygen-sh/community#104] [General]
I've read a great deal of the documentation and am confused on a couple of points so I thought I would seek advice before proceeding.

1. I've created a Product and policy.
2. Fastspring gets the money and details and wants a licence for the product.
3a. Keygen gets customer or
3b. Keygen creates customer
4. <I need to define a licence !!!!!>
5. Keygen gets new licence
6. Fastspring gets licence key code (in a GUI or similar type format)
7. Fastspring gives customer the licence key.

I need to create a licence that will expire and has embedded in it metadata that includes the hashed valued of the name of the club that can use it. This licence key code (in a GUI or similar type format) will be provided to Fastspring. Customer then enters code when requested by my program and my program downloads and installs the licence in a file named "mylicence.lic". Licence is then validated and data extracted so that my program can compare expiry date is not expired and the hash of the name of the club is correct.
The licence expiry date is able to be updated either via subscription (initiated by webhook) or by me.
The licence will be validated at random intervals by my program by contacting Keygen.
The "mylicence.lic" file may be human readable but cannot be tampered with.
The "mylicence.lic" file will be checked for tampering and then expiry and hashed club name will be checked offline on each program startup by reading "mylicence.lic" file.

I think a "Perpetual fallback licence" will be what I want but I am confused by a "Cryptographic licence key" because if I chose such a key then I will need to issue a new key (in a GUI or similar type format) each time I wish to renew it. I don't see a problem in creating a new crypto key when necessary but I don't want to change the licence key code every time.

Am I on the right track?

**Comment by ezekg** [comment]
We created [License Files](https://keygen.sh/blog/announcing-cryptographic-license-files/) for this exact use case, where a signed license key's embedded dataset is planned to change in the future. Instead of a signed key, you can use a plaintext key in combination with a license file. When the expiry changes, simply checkout a new license file. This should ultimately work quite well with your `mylicence.lic` file. We have quite a few examples on our GitHub @keygen_sh of how to utilize license files, in various programming languages, if needed.
